casanovo/casanovo.py

```diff
@@ -363,22 +366,32 @@ def setup_model(
     seed_everything(seed=config["random_seed"], workers=True)
 
     # Download model weights if these were not specified (except when training).
-    if model is None and not is_train:
-        try:
-            model = _get_model_weights()
-        except github.RateLimitExceededException:
-            logger.error(
-                "GitHub API rate limit exceeded while trying to download the "
-                "model weights. Please download compatible model weights "
-                "manually from the official Casanovo code website "
-                "(https://github.com/Noble-Lab/casanovo) and specify these "
-                "explicitly using the `--model` parameter when running "
-                "Casanovo."
+    cache_dir = Path(appdirs.user_cache_dir("casanovo", False, opinion=False))
+    if model is None:
+        if not is_train:
+            try:
```

Comment:
**question:** Maybe with the new abstractions made as part of this PR, the unit tests for the GitHub download functionality can be revisited? (The tests were disabled because they took a long time and it could lead to the GitHub rate being exceeded, but with similar mocking as in your new tests, this might be avoided?)

Additionally, a test of `setup_model` with all its use cases would be relevant, because this is a non-trivial implementation.

Reply:
Sounds good, I added a test for `setup_model` and full mocking of the github functionality used in `get_model_weights` in order to be able to run `test_get_model_weights` without actually touching the github api.